src/runtime/tracing.cpp

```diff
@@ -169,7 +170,7 @@ WEAK int32_t default_trace(void *user_context, const halide_trace_event *e) {
 
         {
             ScopedSpinLock lock(&halide_trace_file_lock);
-            halide_print(user_context, ss.str());
+            halide_print(user_context, (const char *)buffer);
```

Comment:
This should probably have a call to

    halide_msan_annotate_memory_is_initialized(user_context, buffer, strlen(buffer));

to avoid false positives in msan mode.